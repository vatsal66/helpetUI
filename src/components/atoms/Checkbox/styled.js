import styled from 'styled-components';

export const StyledLabel = styled.label`
  position: relative;
  cursor: pointer;
  display: inline-flex;
  padding-left: 35px;
  line-height: 27px;
  font-size: ${({ labelFontSize }) => labelFontSize}px;
  user-select: none;
  color: ${({ labelColor, theme }) => labelColor ?? theme.colors.white};
`;

export const StyledInput = styled.input`
  cursor: pointer;
  opacity: 0;
  height: 0;
  width: 0;

  &:checked ~ span {
    background-color: ${({ checkedColor }) => checkedColor};

    &:after {
      display: block;
    }
  }
`;

export const StyledSpan = styled.span`
  display: flex;
  justify-content: center;
  align-items: center;
  left: 0;
  top: 1px;
  position: absolute;
  height: 24px;
  width: 24px;
  background-color: ${({ uncheckedColor }) => uncheckedColor};
  border-radius: 4px;

  &:after {
    content: '';
    display: none;
    width: 4px;
    height: 8px;
    border: solid ${({ theme }) => theme.colors.white};
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
  }
`;
