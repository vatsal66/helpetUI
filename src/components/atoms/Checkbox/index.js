import React from 'react';
import PropTypes from 'prop-types';

import theme from '@styles/theme';

import { StyledLabel, StyledInput, StyledSpan } from './styled';

const Checkbox = ({
  style,
  labelStyle,
  label,
  labelFontSize,
  labelColor,
  uncheckedColor,
  checkedColor,
  onClick,
}) => (
  <StyledLabel
    style={labelStyle}
    labelFontSize={labelFontSize}
    labelColor={labelColor}
  >
    {label}
    <StyledInput type="checkbox" onClick={onClick} checkedColor={checkedColor} />
    <StyledSpan style={style} uncheckedColor={uncheckedColor} />
  </StyledLabel>
);

Checkbox.propTypes = {
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  labelStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  labelColor: PropTypes.string,
  labelFontSize: PropTypes.number,
  label: PropTypes.string.isRequired,
  uncheckedColor: PropTypes.string,
  checkedColor: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

Checkbox.defaultProps = {
  style: {},
  labelStyle: {},
  labelColor: null,
  labelFontSize: 16,
  uncheckedColor: theme.colors.white,
  checkedColor: theme.colors.primary,
};

export default Checkbox;
