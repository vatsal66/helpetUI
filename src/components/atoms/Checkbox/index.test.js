import React from 'react';
import { fireEvent } from '@testing-library/react';

import renderWithTheme from '@mocks/renderWithTheme';

import Checkbox from './index';

describe('<Checkbox />', () => {
  let component;
  const props = {
    label: 'A Labeled Checkbox',
    onClick: jest.fn(),
  };


  beforeEach(() => {
    component = renderWithTheme(
      <Checkbox {...props} />
    );
  });

  it('should match snapshot', () => {
    const { asFragment } = component;
    expect(asFragment()).toMatchSnapshot();
  });

  it('should fire a onClick function', () => {
    const { getByText } = component;
    fireEvent(
      getByText(props.label),
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );
  });
});
