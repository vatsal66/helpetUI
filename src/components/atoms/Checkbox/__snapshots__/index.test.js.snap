// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should match snapshot 1`] = `
<DocumentFragment>
  <label
    class="sc-AxjAm gBKCgr"
  >
    A Labeled Checkbox
    <input
      class="sc-AxirZ iSLBsk"
      type="checkbox"
    />
    <span
      class="sc-AxiKw ftWzRO"
    />
  </label>
</DocumentFragment>
`;
